nixos/common/detect-hostname-change.nix

```diff
@@ -0,0 +1,43 @@
+# Protection against deploying system closures to the wrong host.
+{
+  config,
+  lib,
+  ...
+}:
+{
+  options.srvos.detect-hostname-change = {
+    enable = lib.mkEnableOption "warn if the hostname changes between deploys" // {
+      default = true;
+    };
+  };
+
+  config = lib.mkIf (config.srvos.detect-hostname-change.enable && config.networking.hostName != "") {
+    system.preSwitchChecks.detectHostnameChange = ''
+      # Ignore if the system is getting installed
+      if [[ ! -e /run/current-system ]]; then
```

Comment:
This handles installing from non-nixos, what about installing from nixos, e.g. nixos-anywhere kexec?

Reply:
Good question. Is there a way to detect the kexec environment?